### PR TITLE
docs(clayui.css): Adds Scss API Documentation

### LINKED ISSUE #4185: Explore the possibility of using sassdoc to add on clayui.com
From #4177

The idea is to try to use some tool that can be used like Sassdoc to extract the Sass documentation from Clay, to expose the variables, mixins, functions that are available on clayui.com in each component's page, just like we do with the API for components in React.

### PR DESCRIPTION
**Pull request by pat270**
This uses sassdoc to generate Scss docs from comments in Clay CSS source. The html gets generated to `static/scss-api/` and I added a link in the main nav to point to it. It uses the default theme provided by sassdoc. We can create our own custom theme if we want.

fixes #4185

### CODE REVIEW

**Comment by pat270** [comment]
`Error:   245:4   error  Expected object keys to be in natural ascending order. 'sideNavigation' should be before 'sidebar'  sort-keys` doesn't look correct to me.

**Comment by matuzalemsteles** [comment]
Wow, I'll look deeper into this tomorrow.

**Comment by pat270** [comment]
Might be looking at the value instead of key?
```
			sideNavigation: 'Side Navigation',
			sidebar: 'Sidebar',
```

**Comment by julien** [comment]
@pat270 what I understand is that since N > b then `sideNavigation` gets sorted before `sidebar`.

**Comment by pat270** [comment]
Thanks @julien! This completely makes sense when you put it that way.

File: clayui.com/gatsby-node.js [hunk 1]
```diff
@@ -3,6 +3,12 @@
...
 
+const express = require('express');
+
 exports.createPages = require('./gatsby/createPages');
 exports.onCreateNode = require('./gatsby/onCreateNode');
 exports.onCreateWebpackConfig = require('./gatsby/onCreateWebpackConfig');
+
+exports.onCreateDevServer = ({app}) => {
+	app.use(express.static('static'));
+};
```
Comment (matuzalemsteles) [inline clayui.com/gatsby-node.js:14]: Hmm this is weird that we have to do this to serve a static file, I thought this was OOTB.

File: clayui.com/plugins/gatsby-plugin-clay-css-tasks/gatsby-node.js [hunk 2]
```diff
@@ -185,13 +186,88 @@ const generateCSSFiles = (pluginOptions) => {
...
+		package: {
+			description: pjson.description,
+			homepage: path.join('..', 'docs', 'get-started', 'index.html'),
+			license: pjson.license,
+			title: 'Clay CSS',
+			version: pjson.version,
+		},
+	};
+
+	sassdoc(path.join(pluginOptions.clayCssSrc, 'scss'), config);
```
Comment (matuzalemsteles) [inline clayui.com/plugins/gatsby-plugin-clay-css-tasks/gatsby-node.js:259]: I had a different idea of this, actually I didn't want to use `sassdoc` to generate the static files but if it was possible to use the API to return the JSON data and so we could use it in different places in the documentation, for example, listed in the tab of `HTML/CSS` of each component. But we can explore that later.

File: clayui.com/src/components/LayoutNav/LayoutNav.js [hunk 3]
```diff
@@ -74,6 +74,18 @@ export default (location) => {
...
 						</li>
+						<li className="nav-item">
+							<a
+								className="nav-link"
+								href="/scss-api/index.html"
+								rel="noopener noreferrer"
+								target="_blank"
+							>
+								<span className="c-inner" tabIndex="-1">
+									{'Scss API'}
```
Comment (matuzalemsteles) [inline clayui.com/src/components/LayoutNav/LayoutNav.js:85]: Shouldn't we call this the `Sass API`?

File: clayui.com/gatsby-node.js (hunk 1 above)
Comment (pat270) [reply clayui.com/gatsby-node.js:14]: @matuzalemsteles this is only in development mode for html files. I found an old issue on the gatsby repo where it errors when using yarn develop. This was the workaround others used. I don't think they fixed it.

File: clayui.com/src/components/LayoutNav/LayoutNav.js (hunk 3 above)
Comment (pat270) [reply clayui.com/src/components/LayoutNav/LayoutNav.js:85]: Probably, I should fix it.

File: clayui.com/gatsby-node.js (hunk 1 above)
Comment (matuzalemsteles) [reply clayui.com/gatsby-node.js:14]: No problem, I'll check this later if we still need it, the our Gatsby version is already getting out of date and they will release the v4 version, and we're still on v2.24 😅

File: clayui.com/plugins/gatsby-plugin-clay-css-tasks/gatsby-node.js (hunk 2 above)
Comment (pat270) [reply clayui.com/plugins/gatsby-plugin-clay-css-tasks/gatsby-node.js:259]: I didn't know they had an API we could use. I have no idea how to use it, but it sounds good.

Comment (matuzalemsteles) [reply clayui.com/plugins/gatsby-plugin-clay-css-tasks/gatsby-node.js:259]: Yeah, this seems to be possible just by calling `sassdoc.parse` which will generate the raw data in a JSON.

**Comment by matuzalemsteles** [comment]
Hey @pat270 I don't see a problem in merging this now as we can deliver great value with it and we can create some issues to integrate better with clayui.com to improve the user experience or just stay like that too if the developers prefer.

**Comment by pat270** [comment]
@matuzalemsteles sounds good, going to merge it.

**Pull request merged**